### DISCUSSION TITLE
[Help wanted!] Let's document all missing interoperable HTML*Element features in 2024!

### DISCUSSION THREAD

**Discussion by teoli2003** [mdn/mdn-community#476] [Content]
In 2024, we would like to document all members (properties, methods) of HTML interfaces (like `HTMLLegendElement` or `HTMLInputElement`)  that are working in every (modern) browser. In other words, we would like one separate reference page for each interoperable HTML feature.

We are looking for help! Thank you in advance!

## Context
MDN Web Docs aims to document every Web platform interface, property, and method. We added thousands of pages over the years and would like to complete our reference documentation of HTML DOM, especially features supported by every browser.

To achieve this, we want to add the 344 missing pages, which is a lot of work. This is why we are looking for help. If you like to write and have some understanding of JavaScript and HTML, we are looking for you!

## How to help

Create a missing page, following our standard, and we'll review it. The list of missing pages is available in this [Google docs](https://docs.google.com/spreadsheets/d/1lvITVZJM8dx8s_Kee5UNI1hDfPNF0gXQwjPop5xIF5E/edit#gid=369545647). Select one and add your name next to it to assign it to you:

![Capture d’écran 2024-01-02 à 07 33 04](https://github.com/mdn/mdn-community/assets/1466293/4a73b58d-5d28-4cfe-b409-441f768173e9)

To create a missing page, create a new page (locally or using the GitHub UI) with the right name, follow the associated template, and listen to the additional instructions from reviewers. Feel free to ask questions here or even in your draft PR:

## Tips

1. **Don't be afraid of making errors**
You can't break anything, and we are here to help. 
1. **Don't overcommit**
Start with one page; there will be comments, and you will learn much from it. Later, you'll be able to create several related pages simultaneously, but don't create too large PRs: they take much longer to review.
3. **Look at existing pages**
MDN Web docs have many reference pages already written: look how they have been written; we are looking for consistency between pages. Typically, you can adapt the structure of an existing similar page. Even if most pages follow the template, some pages differ from it as they are older: the template stays the reference. 
4. **Pay particular attention in writing a helpful example**
 The example must be short but have the necessary context. Where possible, write a live sample. See the [documentation](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Live_samples) about how to do it.
5. **Look at the automatic comments in the PR**
 We have linters adding comments: please fix them (or call for help if you don't understand them). There is also a special comment added to a PR with links to a preview of the page; it also lists flaws that you need to fix. A new page should have 0 flaws. Again, if you are unsure you understand a flaw, feel free to state it in the PR so we can provide assistance. 
 
## Conclusion

This work will allow us to help many beginner developers by providing additional information about these basic features and examples for each.

I would like to thank all of you for your help already.

Quicklinks to the organizational [Google docs](https://docs.google.com/spreadsheets/d/1lvITVZJM8dx8s_Kee5UNI1hDfPNF0gXQwjPop5xIF5E/edit#gid=369545647).

**Comment by Rumyra** [comment]
Thanks for opening this and creating the list @teoli2003 - is it ok if I create a project issue and add it to our roadmap?

**Reply by teoli2003** [reply]
It is absolutely fine! The more people helping here, the better! 🎉

**Reply by Rumyra** [reply]
Added! https://github.com/orgs/mdn/projects/26/views/7

**Comment by teoli2003** [comment]
Let's track our progress: In January 2024, we did 7.53% of the work needed! 😄

_**26 pages of the missing 345 have been added to MDN.**_

Well done! 🎉  Thank you to @skyclouds2001, @ShubhamOulkar, and @OnkarRuikar for their work. ❤️
